src/alphafold_impact/pipelines/data_collection_nsf/nodes.py

```diff
@@ -0,0 +1,136 @@
+"""
+This module contains the functions used to fetch data from the NSF API.
+
+Functions:
+    _create_publication_objects: Create publication objects from a list of paper data.
+    fetch_award_data: Fetch award data for a given award ID.
+    fetch_nsf_data: Fetch NSF data for a given award year.
+"""
+import logging
+from typing import Dict, List
+import requests
+from requests.adapters import HTTPAdapter, Retry
+from joblib import Parallel, delayed
+
+logger = logging.getLogger(__name__)
+
+
+def _create_publication_objects(publication_list: List[str]) -> Dict[str, str]:
+    """
+    Create publication objects from a list of paper data.
+
+    Args:
+        publication_list (List[str]): A list of paper data in the format
+            "journal~year~volume~authors~doi~...".
+
+    Returns:
+        Dict[str, str]: A dictionary of publication objects, where the keys
+            are the NSF PAR IDs and the values are dictionaries
+        containing the publication information (journal, year, volume,
```

Comment:
The line starting with `containing` is not indented to same amount as the other lines